packages/engine-multi/src/util/ensure-payload-size.ts

```diff
@@ -0,0 +1,50 @@
+export const REDACTED_STATE = {
```

Comment:
This part 1: this file has been moved from the worker to the engine, and lightly modified.

Note that it selectively validates payload objects so that it can generate an appropriate "fix"